src/index.js

```diff
@@ -43,58 +57,52 @@ export const gettextFactory = ({
  * gives the `key` and `name` for the token and the `value` gives the value.
  *
  * Singular and plural forms are contained in the same template and delimited by `delimiter`, by
- * default this is the pipe `|` character. The entire template is translated before the singular or
- * plural form is chosen.
+ * default this is the pipe `|` character.
  *
- * Under normal conditions there must be exactly one delimiter in the template. However a custom
- * `choose` method allows any number of delimited forms, and any mapping between factory arguments
- * and the selected form.
+ * The underlying `ngettext` implementation will choose a single form. It is passed the delimited
+ * strings followed by all arguments to the outer function. Validation of arguments is the
+ * concern of `ngettext` and the user must either ensure that the `quantity` is singular or provide
+ * an `ngettext` that performs validation.
  *
  * Where all `substitutions` are `string` then a `string` is returned. Otherwise an `Array` is
  * returned. This allows React elements as substitutions. Elements must have an explicit `key` or
  * they will be assigned one automatically.
  *
- * Failure to pass a `translate` function will not result in an error, but no translation will
- * occur.
+ * Failure to pass a `ngettext` function will not result in an error, but no translation will
+ * occur. This degenerate case supports exactly 2 plural forms which may not suit some developers.
+ * If you need more forms for development then pass a custom `splitPlural` implementation.
  *
+ * Customisation of the `toToken` function is for advanced users only. Make reference to the source
+ * code.
+ *
+ * @throws Error On substitutions with duplicate `name`, or on insufficent plural forms
  * @param {function} [ngettext] Optional translation function, required for translation to occur
  * @param {function} [toToken] Optional custom token inference function, yeilds {name, key, value}
- * @param {string} [delimiter] Optional custom delimiter character
- * @returns {function():function} Factory for a template string interpolator
+ * @param {function} [splitPlural] Optional split string to plural forms, yeilds Array
+ * @param {string} [NODE_ENV] Reserved for testing
+ * @returns {function(quantity:int):function} Factory for a template string interpolator
  */
 export const ngettextFactory = ({
-  ngettext = x => x,
+  ngettext = (s, p, q) => ((typeof q !== 'number') || isNaN(q) || (q === 1) ? s : p),
```

Comment:
That looks like default calues being quite a bit overused here. Why not just extract this as separate function? You probably would like to test it too then